tests/python/pants_test/engine/test_isolated_process.py

```diff
@@ -198,46 +375,40 @@ def test_javac_compilation_example(self):
+    ]))
 
-    scheduler = self.mk_scheduler_in_example_fs([
-      ExecuteProcess.create_in(
-        product_type=ExecuteProcessRequest,
-        input_selectors=(Select(Javac), Select(Snapshot), Select(JavaOutputDir)),
-        input_conversion=process_request_java_args_from_java_sources),
-      SingletonRule(JavaOutputDir, JavaOutputDir('testing')),
-      SingletonRule(Javac, Javac()),
-    ])
-    req = scheduler.product_request(ExecuteProcessRequest, [sources])
-    request = scheduler.execution_request([ExecuteProcessResult], req)
+    scheduler = self.mk_scheduler_in_example_fs(create_javac_compile_rules())
 
-    root_entries = scheduler.execute(request).root_products
-    self.assertEquals(1, len(root_entries))
-    state = self.assertFirstEntryIsReturn(root_entries, scheduler, request)
-    execution_result = state.value
-    self.assertEqual(0, execution_result.exit_code)
+    request = JavacCompileRequest(
+      javac_compile_command=JavacCompileCommand(),
+      javac_sources=javac_sources,
+    )
+
+    results = self.execute(scheduler, JavacCompileResult, request)
+    self.assertEquals(1, len(results))
+    javac_compile_result = results[0]
+    self.assertEqual(0, javac_compile_result.exit_code)
     # TODO: Test that the output snapshot is good
 
   def test_javac_compilation_example_rust_failure(self):
-    sources = PathGlobs.create('', include=['scheduler_inputs/src/java/simple/Broken.java'])
-
-    scheduler = self.mk_scheduler_in_example_fs([
-      ExecuteProcess.create_in(
-        product_type=ExecuteProcessRequest,
-        input_selectors=(Select(Javac), Select(Snapshot), Select(JavaOutputDir)),
-        input_conversion=process_request_java_args_from_java_sources),
-      SingletonRule(JavaOutputDir, JavaOutputDir('testing')),
-      SingletonRule(Javac, Javac()),
-    ])
-    req = scheduler.product_request(ExecuteProcessRequest, [sources])
-    request = scheduler.execution_request([ExecuteProcessResult], req)
-
-    root_entries = scheduler.execute(request).root_products
-    self.assertEquals(1, len(root_entries))
-    state = self.assertFirstEntryIsReturn(root_entries, scheduler, request)
-    execution_result = state.value
-    self.assertEqual(1, execution_result.exit_code)
-    self.assertIn("NOT VALID JAVA", execution_result.stderr)
+    javac_sources = JavacSources(PathGlobs.create('', include=[
+      'scheduler_inputs/src/java/simple/Broken.java',
+    ]))
+
+    scheduler = self.mk_scheduler_in_example_fs(create_javac_compile_rules())
+
+    request = JavacCompileRequest(
+      javac_compile_command=JavacCompileCommand(),
+      javac_sources=javac_sources,
+    )
+
+    results = self.execute(scheduler, JavacCompileResult, request)
+    self.assertEquals(1, len(results))
+    javac_compile_result = results[0]
+    self.assertEqual(1, javac_compile_result.exit_code)
+    self.assertIn("NOT VALID JAVA", javac_compile_result.stderr)
 
   def test_failed_command_propagates_throw(self):
```

Comment:
Feel free to delete this, and the entire `SnapshottedProcess` everything, it feels unnecessary now

Reply:
Everything related to SnapshottedProcess was deleted, a TODO was left on JavacCompileResult so it's clear where to edit once we can make process output file snapshots.